src/org/labkey/test/util/search/SearchAdminAPIHelper.java

```diff
@@ -28,26 +29,54 @@
 import java.util.Map;
 
 import static org.junit.Assert.assertEquals;
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_PAGE;
 
 public abstract class SearchAdminAPIHelper
 {
+
+    public static void waitForIndexer()
+    {
+        waitForIndexer(WAIT_FOR_PAGE);
+    }
+
     /**
      * Wait for search indexer to be idle via SearchController.WaitForIndexerAction
      */
     @LogMethod(quiet = true)
-    public static void waitForIndexer()
+    public static void waitForIndexer(int timeout)
     {
         // Invoke a special server action that waits until all previous indexer tasks are complete
-        int response = WebTestHelper.getHttpResponse(WebTestHelper.buildURL("search", "waitForIndexer")).getResponseCode();
-        assertEquals("WaitForIndexer action timed out", HttpStatus.SC_OK, response);
+        var cmd = new PostCommand("search", "waitForIndexer");
+        cmd.setTimeout(timeout);
+        executeWaitForIndexer(cmd);
     }
 
-    @LogMethod(quiet = true)
     public static void waitForIndexerBackground()
+    {
+        waitForIndexerBackground(WAIT_FOR_PAGE);
+    }
+
+    @LogMethod(quiet = true)
+    public static void waitForIndexerBackground(int timeout)
     {
         // Invoke a special server action that waits until all previous indexer tasks are complete, even wait for background indexing tasks to complete (e.g. deleteContainer)
-        int response = WebTestHelper.getHttpResponse(WebTestHelper.buildURL("search", "waitForIndexer", Map.of("priority","background"))).getResponseCode();
-        assertEquals("WaitForIndexer action timed out", HttpStatus.SC_OK, response);
+        var cmd = new PostCommand("search", "waitForIndexer");
+        cmd.setTimeout(timeout);
+        cmd.setParameters(Map.of("priority","background"));
+
+        executeWaitForIndexer(cmd);
+    }
+
+    private static void executeWaitForIndexer(PostCommand cmd)
+    {
+        try
+        {
+            var response = cmd.execute(WebTestHelper.getRemoteApiConnection(), null);
+            assertEquals("WaitForIndexer action timed out", HttpStatus.SC_OK, response.getStatusCode());
+        } catch (Exception cmdException)
+        {
+            throw new RuntimeException("an error occurred while waiting for search indexing to finish", cmdException);
+        }
```

Comment:
I wouldn't bother breaking `executeWaitForIndexer` out to a separate method. It's short and only used in one place.

Reply:
Sorry. You had it right the first time. I misread the code. Didn't realize that method was getting called twice.

Reply:
hahahahahahaa ok I'll put it back